src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -453,7 +458,10 @@ private static void DeserializeFrameworkTypeValue(CodeWriter writer, CodeWriterD
         {
             bool includeFormat = false;
 
-            writer.Append($"{element}.");
+            if (frameworkType == typeof(ETag))
+                writer.Append($"new {typeof(ETag)}({element}.");
```

Comment:
Sorry, should've been more specific. I was talking about something like:

```
if (frameworkType == typeof(ETag))
{
	writer.Append($"new {typeof(ETag)}({element}.GetString())");
	return;
}
```

This would keep all ETag in one place instead of behind multiple ifs.

Reply:
Ah, yes makes sense! I don't know why I didn't do this before :(